depthai_nodes/node/utils/nms.py

```diff
@@ -54,6 +56,58 @@ def nms_detections(detections: List[dai.ImgDetection], conf_thresh=0.3, iou_thre
     return final_detections
 
 
+def nms_detections_extended(
+    detections: List[ImgDetectionExtended], conf_thresh=0.3, iou_thresh=0.4
+):
+    """Applies Non-Maximum Suppression (NMS) on a list of dai.ImgDetection objects.
+
+    @param detections: List of dai.ImgDetection objects. @type
+    detections: list[dai.ImgDetection] @param conf_thresh: Confidence
+    threshold for filtering boxes. @type conf_thresh: float @param
+    iou_thresh: IoU threshold for Non-Maximum Suppression (NMS). @type
+    iou_thresh: float
+
+    @return: A list of dai.ImgDetection objects after applying NMS.
+    @rtype: list[dai.ImgDetectionExtended]
+    """
+    if len(detections) == 0:
+        return []
+
+    # Filter out detections below confidence threshold
+    detections = [det for det in detections if det.confidence >= conf_thresh]
+    if len(detections) == 0:
+        return []
```

Comment:
You can merge the first and second condition into one.

This
```
    if len(detections) == 0:
        return []

    # Filter out detections below confidence threshold
    detections = [det for det in detections if det.confidence >= conf_thresh]
    if len(detections) == 0:
        return []
```
is equivalent to this
```
detections = [det for det in detections if det.confidence >= conf_thresh]
    if len(detections) == 0:
        return []
```